lib/http2/connection.c

```diff
@@ -606,6 +606,10 @@ static int handle_priority_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *fram
         return H2O_HTTP2_ERROR_PROTOCOL;
     }
 
+    /* Ignore PRIORITY frames for pushed idle streams */
+    if (!(frame->stream_id & 1) && (frame->stream_id > conn->push_stream_ids.max_open))
+            return 0;
```

Comment:
How about merging this with the `if` statement you added below?

The merits are:
* the if statements, combined would become simple, i.e.: `if (h2o_http2_stream_is_push(frame->stream_id)) { /* Ignore PRIORITY frames for pushed streams in idle or closed state */ return 0; } else { ... }`
* the check would move away from the hot path